src/app/backend/resource/customresourcedefinition/v1/common.go

```diff
@@ -123,17 +123,17 @@ func isServed(crd apiextensions.CustomResourceDefinition) bool {
 
 func removeNonServedVersions(crd apiextensions.CustomResourceDefinition) apiextensions.CustomResourceDefinition {
 	versions := append(crd.Spec.Versions)
-	for i, version := range crd.Spec.Versions {
-		if len(versions) == 1 && !version.Served {
-			versions = make([]apiextensions.CustomResourceDefinitionVersion, 0)
-			break
-		}
-
-		if !version.Served {
-			versions = append(versions[:i], versions[i+1:]...)
+	if len(versions) == 1 && !versions[0].Served {
```

Comment:
Honestly, I am not sure why I have complicated it so much. Let's simply reverse the logic here and start with an empty array and add only served versions. Thanks for reporting this!
 
```go
	versions := make([]apiextensions.CustomResourceDefinitionVersion, 0)

	for _, version := range crd.Spec.Versions {
		if version.Served {
			versions = append(versions, version)
		}
	}

	crd.Spec.Versions = versions
	return crd
```

Reply:
yep, that would work just fine. This is actually my first thought of fix, but I thought you may have some other considerations to use an in-place like removal logic. Then I followed your pattern. 😆